src/main/java/uk/nhs/hee/tis/trainee/credentials/dto/PlacementCredentialDto.java

```diff
@@ -117,4 +118,9 @@ public String getTisId() {
   public UUID getUniqueIdentifier() {
     return uniqueIdentifier;
   }
+
+  @JsonIgnore
+  public int calculatePlacementHashCode() {
+    return Objects.hash(specialty, grade, nationalPostNumber, employingBody, site, startDate, endDate);
```

Comment:
This technique is not guaranteed to give consistent hash values across Java versions, or even across separate executions. We could, potentially, get different results across different task instances running in parallel.